src/ContainerCheckpointModal.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Modal } from 'patternfly-react';
+import { Button, Checkbox } from '@patternfly/react-core';
+import cockpit from 'cockpit';
+import * as utils from './util.js';
+
+const _ = cockpit.gettext;
+
+class ContainerCheckpointModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            keep: false,
+            leaveRunning: false,
+            tcpEstablished: false,
+            ignoreRootFS: false
+        };
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(checked, event) {
+        if (event.target.type === "checkbox")
+            this.setState({ [event.target.name]: event.target.checked });
+    }
+
+    render() {
+        return (
+            <Modal show={this.props.selectContainerCheckpointModal}>
+                <Modal.Header>
+                    <Modal.Title>
+                        {cockpit.format(_("Checkpoint container $0"),
+                                        utils.truncate_id(this.props.containerWillCheckpoint.Id))}
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <div className="ct-form">
+                        <Checkbox label={_("Keep all temporary checkpoint files")} id="keep" name="keep"
+                                  isChecked={this.state.keep} onChange={this.handleChange} />
+                        <Checkbox label={_("Leave the container running after writing checkpoint to disk")}
```

Comment:
I would make **the** into **this** for slightly more clarity. Was also considering if we should drop the **container** part, but even though that makes the text shorter, it might make it more unclear.

Reply:
"Leave" (in the meaning used here) is a transitive verb, I think it should have an object. "This container" seems to me like implying that there are another containers in the context, which isn't the case (the dialog mentions a specific container in the heading). In my opinion using "the container" is better than "this container" or dropping the object altogether.

Reply:
Could also make it `Leave container running after writing checkpoint to disk`

Reply:
Missed the last part apparently. Dropping the object all together sounds like a good idea. As it's implied by the context.

Reply:
I actually didn't mean that by the last part, but I've changed my mind about it - considering I dropped the "container" object in another case (above) I now don't see a problem with doing it here.